examples/go-sdk/main.go

```diff
@@ -0,0 +1,37 @@
+package main
+
+import (
+	"fmt"
+
+	configu "github.com/configu/configu/go"
+)
+
+func main() {
+	store := configu.InMemoryStore{}
+	set, err := configu.NewConfigSet("test")
```

Comment:
@jamesL92 For the instance creation, is the New prefix necessary? If it isn't, I'd recommend renaming it to simply ConfigSet, ConfigSchema, and so on.

Reply:
We _can_ do without it, however there are tradeoffs to consider. It's even possible with the current implementation.

This is typically the way to handle custom constructor logic in Go - e.g. for `ConfigSet`s verifying that the path is valid and for `ConfigSchema`s ensuring that the config schema uses the correct extension. The other way would be to have this logic in a separate method:
```go
set := configu.ConfigSet{"test"}
err := set.ValidatePath()
```

I would recommend sticking with the `NewConfigSet` semantics, since:
- it's more in line (logically) with other SDKs
- the user doesn't have to "remember" to call a separate method for such custom logic.

Happy to go with either approach, let me know if you'd like me to change this and I'll do so accordingly :+1: